### PR TITLE
 libvirt_pci_passthrough: refactor tests to use `host_pf_filter` for device pass-through.

### PR DESCRIPTION
**Pull request by santwana**
The PciAssignable class uses the "host_pf_filter" for the PCI Devices
to be pass-through'ed, and the same can be used in the test-case.
Now the user also have the flexibilty to create the desired no.of Virtual Functions,
for each Physical Function incase of SR-IOV.

Removed the device_name params for comparision of the PCI devices
before/after passthrough.

Signed-off-by: Santwana Samantray <santwana@linux.vnet.ibm.com>

### CODE REVIEW

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 1]
```diff
@@ -192,4 +171,6 @@ def run(test, params, env):
         backup_xml.sync()
         # For SR-IOV , VF's should be cleaned up in the post-processing.
         if sriov:
-            sriov_setup.release_devs()
+            if obj.get_vfs_count() != 0:
+                for clean in pci_ids:
+                    obj.set_vf(clean)
```
Comment (sathnaga) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: minor though, for readability, we can pass  `vf_no="0" ` explicitly to say we are reseting/cleaning...

I would prefer having a
```
def  reset_vf(pci_pf):
     set_vf(pci_pf, vf_no="0")
```

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 2]
```diff
@@ -192,4 +171,6 @@ def run(test, params, env):
         backup_xml.sync()
         # For SR-IOV , VF's should be cleaned up in the post-processing.
         if sriov:
-            sriov_setup.release_devs()
+            if obj.get_vfs_count() != 0:
+                for clean in pci_ids:
```
Comment (sathnaga) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: `for pci_pf in pci_ids:` would be better

File: libvirt/tests/cfg/libvirt_pci_passthrough.cfg [hunk 3]
```diff
@@ -12,8 +12,9 @@
             # a network device. We will attach this
             # device to guest. Then this network device
             # will be unavailable on host.
-            # E.g: pci_0000_05_00_0
-            libvirt_pci_net_dev_label = "ENTER.YOUR.PCI.LABEL"
+            # E.g: 0000:05:00.0
+            pf_filter = "ENTER.YOUR.PCI.LABEL"
+            vf_filter = "Virtual Function"
```
Comment (sathnaga) [inline libvirt/tests/cfg/libvirt_pci_passthrough.cfg]: move `vf_filter` and `nvf` param under SRIOV variant, it makes more sense there

File: libvirt/tests/cfg/libvirt_pci_passthrough.cfg [hunk 4]
```diff
@@ -24,6 +25,9 @@
             libvirt_pci_net_dev_name = "ENTER.YOUR.DEVICE.NAME"
             # Enter netmask in CIDR notation
             libvirt_pci_net_mask = "ENTER.YOUR.NETMASK"
+            # Enter the no.of Virtual Function's to be created
+            # for each Physical Function.
+            nvf = 3
```
Comment (sathnaga) [inline libvirt/tests/cfg/libvirt_pci_passthrough.cfg]: this param can be little expanded, like, `number_vfs = 3`

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 5]
```diff
@@ -66,37 +68,23 @@ def run(test, params, env):
...
         nic_list_before = output.splitlines()
+        obj = PciAssignable(pf_filter_re=pf_filter,
+                            vf_filter_re=vf_filter)
+        # get all functions id's
+        pci_ids = obj.get_same_group_devs(pf_filter)
+        pci_devs = []
+        for val in pci_ids:
+            temp = val.replace(":", "_")
+            pci_devs.extend(["pci_"+temp])
         if sriov:
```
Comment (sathnaga) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: change is not part of this patch, though we can handle now choosing sriov by `number_vfs` or `vf_filter` instead of additional param.

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 6]
```diff
@@ -177,7 +156,7 @@ def run(test, params, env):
                 logging.info(o_ping)
                 if s_ping != 0:
                     err_msg = "Ping test fails, error info: '%s'"
-                    test.fail(err_msg % o_ping)
+                    logging.debug(err_msg % o_ping)
```
[the remark this answers is not in the document]
Comment (santwana) [reply libvirt/tests/src/libvirt_pci_passthrough.py]: @sathnaga  Under certain circumstances if the network setting's are not proper, its better to print the error log, rather than failing the test-case.

File: libvirt/tests/src/libvirt_pci_passthrough.py (hunk 1 above)
Comment (santwana) [reply libvirt/tests/src/libvirt_pci_passthrough.py]: @sathnaga Addressed the changes suggested, kindly review.

File: libvirt/tests/cfg/libvirt_pci_passthrough.cfg (hunk 3 above)
Comment (santwana) [reply libvirt/tests/cfg/libvirt_pci_passthrough.cfg]: @sathnaga changed as suggested.

File: libvirt/tests/cfg/libvirt_pci_passthrough.cfg (hunk 4 above)
Comment (santwana) [reply libvirt/tests/cfg/libvirt_pci_passthrough.cfg]: @sathnaga changed as suggested.

**Comment by sathnaga** [comment]
On Thu, May 30, 2019 at 02:25:18AM -0700, santwana wrote:
> santwana commented on this pull request.
> 
> 
> 
> > @@ -177,7 +156,7 @@ def run(test, params, env):
>                  logging.info(o_ping)
>                  if s_ping != 0:
>                      err_msg = "Ping test fails, error info: '%s'"
> -                    test.fail(err_msg % o_ping)
> +                    logging.debug(err_msg % o_ping)
> 
> @sathnaga  Under certain circumstances if the network setting's are not proper, its better to print the error log, rather than failing the test-case. 

@santwana in that case, better to control via an param like `ignore_status`, and decide via config, as it is a generic testcase and changing test case failure
statement is not a good idea.

**Comment by santwana** [comment]
@sathnaga reverted back to the test.fail , please review. Thanks!!

**Comment by sathnaga** [comment]
@santwana you might want to squash the commits, if not would expect you to revert the test.fail from first commit, coz that is the one changes to debug in first place.

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 7]
```diff
@@ -144,24 +123,23 @@ def run(test, params, env):
...
             for val in range(len(nic_list)):
                 bus_info.append(str(nic_list[val]).split(' ', 1)[0])
                 nic_list[val] = str(nic_list[val]).split(' ', 1)[0][:-2]
             bus_info.sort()
             if not sriov:
                 # check all functions get same iommu group
                 if len(set(nic_list)) != 1:
-                    test.fail("Multifunction Device passthroughed but "
-                              "functions are in different iommu group")
+                    logging.debug("Multifunction Device passthroughed but "
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: please raise the fail()

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 8]
```diff
@@ -28,19 +27,21 @@ def run(test, params, env):
         5. Compare the result of 'fdisk -l' before and after
             attaching storage pci device to guest.
     """
+    def reset_vf(pci_pf):
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: this method is not required, please call set_vf() directly.

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 9]
```diff
@@ -144,24 +123,23 @@ def run(test, params, env):
         session = vm.wait_for_login()
         # The Network configuration is generic irrespective of PF or SRIOV VF
         if device_type == "NIC":
-            output = session.cmd_output("lspci -nn")
+            output = session.cmd_output("lspci -D")
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: `vm.get_pci_devices()`

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 10]
```diff
@@ -50,53 +51,40 @@ def run(test, params, env):
...
+        test.cancel("Please enter your Storage Adapter details for test.")
     fdisk_list_before = None
     vmxml = VMXML.new_from_inactive_dumpxml(vm_name)
     backup_xml = vmxml.copy()
     if device_type == "NIC":
         if not vm.is_alive():
             vm.start()
         session = vm.wait_for_login()
-        output = session.cmd_output("lspci -nn")
+        output = session.cmd_output("lspci -D")
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: same here

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 11]
```diff
@@ -50,53 +51,40 @@ def run(test, params, env):
...
         nic_list_before = output.splitlines()
+        obj = PciAssignable(pf_filter_re=pf_filter,
+                            vf_filter_re=vf_filter)
+        # get all functions id's
+        pci_ids = obj.get_same_group_devs(pf_filter)
+        pci_devs = []
+        for val in pci_ids:
+            temp = val.replace(":", "_")
+            pci_devs.extend(["pci_"+temp])
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: we discussed to avoid these code to convert `:` and `_`, please have a look

Comment (santwana) [reply libvirt/tests/src/libvirt_pci_passthrough.py]: @balamuruhans I have already taken care of the config changes from pci_ to : format, the existing code is required for the NodedevXML .

**Comment by santwana** [comment]
@balamuruhans Implemented the changes as suggested, kindly review. Thanks!

**Comment by balamuruhans** [comment]
```
